.github/workflows/label-prs.yml

```diff
@@ -0,0 +1,95 @@
+name: Label PR
+
+on:
+  pull_request:
+    types: [assigned, opened, edited, synchronize, reopened]
+
+jobs:
+  validate-branch-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate branch name
+        env:
+          HEAD_REF: ${{ github.event.pull_request.head.ref }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const BRANCH_REGEX = /^main|develop|(ci|chore|docs|feat|fix|perf|refactor|revert|style)\/\d*(?:[-](?![-])\w*)*$/;
+            const ERROR_MSG =
+              'This branch name does not follow our conventions.' +
+              '\n' +
+              'Rename it with "git branch -m <current-name> <new-name>"' +
+              '\n' +
+              'Here are some example branch names that are accepted: "fix/123-description", "feat/123-longer-description", "chore/123", "main", "develop".';
+
+            const isValidBranchName = new RegExp(BRANCH_REGEX).test(process.env.HEAD_REF);
+            if (!isValidBranchName) {
+              console.error(ERROR_MSG);
+              process.exit(1);
+            }
+
+  label-pr:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - name: Label PR
+        env:
+          HEAD_REF: ${{ github.event.pull_request.head.ref }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            if (process.env.HEAD_REF === 'main' || process.env.HEAD_REF === 'develop') {
```

Comment:
It might be better if we could extract this script to a separate file in `.circleci/scripts`. Then it'd get proper syntax highlighting, it could be linted, etc.